samples/bluetooth/hci_uart/src/main.c

```diff
@@ -150,8 +150,7 @@ static void rx_isr(void)
 				 * interrupt. On failed allocation state machine
 				 * is reset.
 				 */
-				buf = bt_buf_get_tx(BT_BUF_H4, K_NO_WAIT,
-						    &type, sizeof(type));
+				buf = bt_buf_get_tx(type, K_NO_WAIT, NULL, 0);
```

Comment:
Consider updating the documentation for the public function `bt_buf_get_tx` that `data` can be NULL

Reply:
> Consider updating the documentation for the public function `bt_buf_get_tx` that `data` can be NULL

I can do that, but it's always been the case, i.e. this PR doesn't change anything in that regard.

Reply:
That's not quite true; some of the code you removed accessed the `data` parameter unconditionally : 

```c
	case BT_BUF_H4:
		if (IS_ENABLED(CONFIG_BT_HCI_RAW_H4) &&
		    raw_mode == BT_HCI_RAW_MODE_H4) {
			uint8_t h4_type = ((uint8_t *)data)[0];
```

So previously, as least for `BT_BUF_H4`, `data` could not be NULL

Reply:
Right, in the H:4 encoding case (for which the experimental USB feature was the only user) that's indeed true. However, that feature was not being enabled anywhere, so for all other cases you could pass NULL. Anyway, I'll add a note to the documentation.

Reply:
Added note to the documentation. Btw, this is a purely HCI raw API (something that the `buf.h` documentation also doesn't mention anything about. I think we should get rid of APIs like this in order to simplify the HCI interface even further.